### PR TITLE
Add authentication to For Sponsor pages and refactored to remove state hooks from App.js

### PR DESCRIPTION
**Pull request by M-Lieberman**

### CODE REVIEW

**Comment by M-Lieberman** [comment]
closing pull request and recreating after fixing merge conflicts

File: src/App.js [hunk 1]
```diff
@@ -52,36 +55,75 @@ export default function App() {
...
   ]);
 
-  // TODO fetch sponsorId from currentUser context
-  const currentUser = {
-    uid: "b19dcdc9-1547-11eb-9ed1-0a7222284ed8",
-    userType: "sponsor"
-  }
-  // Fetch the requests from the API
-  const [fundingRequests, setRequests] = useState([]);
+  // const [detailsCharity, setdetailsCharity] = useState([
```
Comment (leireulacia) [inline src/App.js:58]: You can delete lines from  58 to  126

**Review by leireulacia — CHANGES_REQUESTED** [review]
Please, review my comments in App.js and SponsorRequestDetails.js

File: src/components/sponsorrequestdetails/SponsorRequestDetails.js [hunk 2]
```diff
@@ -3,39 +3,50 @@ import Container from 'react-bootstrap/Container';
...
     const handleClose = () => setShow(false);
     const handleShow = () => setShow(true);
-
-
     const handleClick = () => {
         setShow(false);
         history.push("/ForSponsorsPending");
     }
 
+    const currentUser = {
```
[the remark this answers is not in the document]
Comment (M-Lieberman) [reply src/components/sponsorrequestdetails/SponsorRequestDetails.js:37]: yes, planning on moving it in my next branch

**Review by leireulacia — APPROVED** [review]
Looks great Mark!

**Pull request merged**